### PR TITLE
Fix metering for internal events

### LINKED ISSUE #695: Non-diagnostic events are currently unmetered
This bug was previously about the event system taking quadratic time due to using MeteredVec. In the meantime this has been changed to be a non-metered standard vector, but this needs to be finished to be metered.

### PR DESCRIPTION
**Pull request by jayz22**
### What

Fixes #695 

### Why

[TODO: Why this change is being made. Include any context required to understand the why.]

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: soroban-env-host/src/events/internal.rs [hunk 1]
```diff
@@ -141,15 +142,24 @@ pub(crate) struct InternalEventsBuffer {
...
 impl InternalEventsBuffer {
     // Records an InternalEvent
-    pub fn record(&mut self, e: InternalEvent, _budget: &Budget) -> Result<(), HostError> {
-        //TODO:Add metering for non-diagnostic events
+    pub fn record(&mut self, e: InternalEvent, budget: &Budget) -> Result<(), HostError> {
+        // Metering: we use the cost of instantiating a size=1 `Vec` as an estimate for the cost
+        // `Vec.push(event)`. Because the buffer length may be different on different instances
+        // due to diagnostic events and we need a deterministic cost across all instances,
+        // the cost needs to be amortized and buffer size-independent.
+        metered_clone::charge_container_bulk_init_with_elts::<
```
Comment (sisuresh) [inline soroban-env-host/src/events/internal.rs:150]: Does this also charge for Diagnostic events?

Comment (jayz22) [reply soroban-env-host/src/events/internal.rs]: Oh you are right, good catch! 
Fixed.

Comment (jayz22) [reply soroban-env-host/src/events/internal.rs]: I've added a couple of tests.

**Comment by jayz22** [comment]
@sisuresh I think this one is ready to be merged.

**Review by sisuresh — APPROVED** [review]

**Pull request merged**